src/Evolution/Systems/Cce/Equations.cpp

```diff
@@ -0,0 +1,268 @@
+    const SpinWeighted<ComplexDataVector, 1>& eth_r_divided_by_r,
+    const SpinWeighted<ComplexDataVector, 0>& k,
+    const SpinWeighted<ComplexDataVector, 0>& one_minus_y,
+    const SpinWeighted<ComplexDataVector, 0>& r) noexcept {
+  *script_aj =
+      0.25 *
+      (conj(ethbar_ethbar_j) -
+       0.25 * (4.0 + eth_ethbar_j_jbar - j * conj(eth_ethbar_j)) /
+           (k * (1.0 + j * conj(j))) +
+       (3.0 - eth_ethbar_beta -
+        conj(j) * eth_ethbar_j * (1.0 - 0.25 / (1.0 + j * conj(j)))) /
+           k +
+       conj(ethbar_j) * (2.0 * eth_beta +
+                         0.5 *
+                             (j * conj(eth_j_jbar) -
+                              ethbar_j * (2.0 * (1.0 + j * conj(j)) - 1.0)) /
+                             (k * (1.0 + j * conj(j))) -
+                         q));
+
+  *script_bj =
+      0.25 * (2.0 * dy_w -
+              conj(j) * eth_u * (conj(j) * dy_j + j * conj(dy_j)) / k +
+              1.0 / r + u * ethbar_j * conj(dy_j) -
+              0.5 * u * conj(eth_j_jbar) * (conj(j) * dy_j + j * conj(dy_j)) /
+                  (1.0 + j * conj(j)) +
+              conj(u) * (conj(ethbar_jbar_dy_j) - j * conj(ethbar_dy_j))) +
+      one_minus_y * 0.25 *
+          (du_r_divided_by_r * dy_j *
+               (conj(j) * (conj(j) * dy_j + j * conj(dy_j)) /
+                    (1.0 + j * conj(j)) -
+                2.0 * conj(dy_j)) -
+           w * (dy_j * conj(dy_j) -
+                0.25 * square((conj(j) * dy_j + j * conj(dy_j))) /
+                    (1.0 + j * conj(j)))) +
+      square(one_minus_y) * 0.125 *
+          (0.25 * square(conj(j) * dy_j + j * conj(dy_j)) /
+               (1.0 + j * conj(j)) -
+           dy_j * conj(dy_j)) /
+          r;
+
+  *script_cj = 0.5 * ethbar_j * k * (eth_beta - 0.5 * q);
+
+  *regular_integrand_for_h =
+      j * (*script_bj + conj(*script_bj)) -
+      0.5 * (eth_ubar_dy_j + u * ethbar_dy_j +
+             u * dy_j * conj(eth_r_divided_by_r)) +
+      0.5 * exp_2_beta / r *
+          (*script_cj + square(j) / (1.0 + j * conj(j)) * conj(*script_cj) -
+           j * (*script_aj + conj(*script_aj)) + eth_eth_beta - 0.5 * eth_q +
+           0.25 * (conj(ethbar_jbar_q_minus_2_eth_beta) - j * conj(ethbar_q)) /
+               k +
+           square(eth_beta - 0.5 * q)) -
+      dy_j * 0.5 *
+          (conj(j) * eth_u / k - j * k * conj(eth_u) +
+           0.5 * (1.0 + j * conj(j)) * (conj(ethbar_u) - ethbar_u) +
+           conj(u) * eth_r_divided_by_r - w) +
+      conj(dy_j) * (0.5 * j * eth_u * (j * conj(j) / k) -
+                    0.25 * square(j) * (ethbar_u - conj(ethbar_u))) +
+      one_minus_y *
+          (0.5 * (dy_dy_j * (w + 2.0 * du_r_divided_by_r) - dy_j / r) +
+           0.5 * dy_j * (dy_w + 1.0 / r)) +
+      square(one_minus_y) * 0.25 * dy_dy_j / r;
+}
+
+void ComputeBondiIntegrand<Tags::LinearFactor<Tags::BondiH>>::apply_impl(
+    const gsl::not_null<SpinWeighted<ComplexDataVector, 0>*>
+        linear_operator_for_h,
```

Comment:
Tag is `LinearFactor<H>` but variable name is `linear_operator_for_h` ... is this intentional?